packages/marshal/test/test-rankOrder.js

```diff
@@ -0,0 +1,227 @@
+// @ts-check
+import { test } from './prepare-test-env-ava.js';
+
+import { Far } from '../src/make-far.js';
+import { makeTagged } from '../src/makeTagged.js';
+import {
+  FullRankCover,
+  compareRank,
+  isRankSorted,
+  makeRankSorted,
+  getIndexCover,
+  getPassStyleCover,
+  assertRankSorted,
+} from '../src/rankOrder.js';
+
+const { quote: q } = assert;
+
+/**
+ * The only elements with identity. Everything else should be equal
+ * by contents.
+ */
+const alice = Far('alice', {});
+const bob = Far('bob', {});
+const carol = Far('carol', {});
+
+/**
+ * An unordered copyArray of some passables
+ */
+const sample = harden([
+  makeTagged('copySet', [
+    ['b', 3],
+    ['a', 4],
+  ]),
+  'foo',
+  3n,
+  'barr',
+  undefined,
+  [5, { foo: 4 }],
+  2,
+  null,
+  [5, { foo: 4, bar: null }],
+  bob,
+  0,
+  makeTagged('copySet', [
+    ['a', 4],
+    ['b', 3],
+  ]),
+  NaN,
+  true,
+  undefined,
+  [5],
+  alice,
+  [],
+  Symbol.for('foo'),
+  new Error('not erroneous'),
+  Symbol.for('@@foo'),
+  [5, { bar: 5 }],
+  Symbol.for(''),
+  false,
+  carol,
+  -0,
+  {},
+  [5, undefined],
+  -3,
+  makeTagged('copyMap', [
+    ['a', 4],
+    ['b', 3],
+  ]),
+  true,
+  'bar',
+  [5, null],
+  new Promise(() => {}), // forever unresolved
+  makeTagged('nonsense', [
+    ['a', 4],
+    ['b', 3],
+  ]),
+  Symbol.isConcatSpreadable,
+  [5, { foo: 4, bar: undefined }],
+  Promise.resolve('fulfillment'),
+  [5, { foo: 4 }],
+]);
+
+const rejectedP = Promise.reject(new Error('broken'));
+rejectedP.catch(() => {}); // Suppress unhandled rejection warning/error
+
+/**
+ * The correctly stable rank sorting of `sample`
+ */
+const sortedSample = harden([
+  null,
+  false,
+  true,
+  true,
+
+  // -0 is equivalent enough to 0. NaN after all numbers.
```

Comment:
I think the linter stripped out your `-0`.  Maybe try `Number('-0')`?

Reply:
I checked. `yarn lint-fix` does not revise `-0`. So I just changed to `-0` without the `Number`.